documentation/writing_tests.md

```diff
@@ -14,6 +20,139 @@
 
 ## Recommended tools
 
+Writing tests for JavaScript and TypeScript requires testing tools, such as a test framework, an assertion library and a way to bundle and run these tests on various environments. The JavaScript community has many overlapping tools that one could pick to fulfill any of these tasks. To ensure that the testing experience across the Azure SDK for JS/TS is consistent and reliable,
+we've picked the following external testing tools:
+
+- [Mocha](https://www.npmjs.com/package/mocha), which offers us a well known and stable test framework for both NodeJS and browser tests.
+- [Chai](https://www.npmjs.com/package/chai), a well known assertion library for Node and the browser.
+- [Rollup](https://www.npmjs.com/package/rollup), to bundle JavaScript for different environments, which helps us write a single TypeScript source and trust it will compile correctly for Node and the browser.
+- [Karma](https://www.npmjs.com/package/karma), which allows us to run our tests in multiple browsers.
+
+We've also come up with our own internal tools, which follow:
+
+- The [Recorder](https://github.com/Azure/azure-sdk-for-js/tree/master/sdk/test-utils/recorder) is a tool that helps us run our live tests against static recordings obtained from a previous successful run, which aims to ensure that our code hasn't changed while benefitting from not having to reach out to real live services.
+
+Keep in mind that, over time, we might continue to expand these lists.
```

Comment:
nit: not sure if this sentence is necessary

Reply:
I'll remove it!